crates/biome_js_analyze/src/assist/source/organize_imports/import_source.rs

```diff
@@ -0,0 +1,379 @@
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+enum PathComponent<'a> {
+    /// See [Component::Prefix]
+    Prefix(std::path::PrefixComponent<'a>),
+    /// See [Component::RootDir]
+    RootDir,
+    /// See [Component::ParentDir]
+    ParentDir(usize),
+    /// See [Component::CurDir]
+    CurDir,
+    /// See [Component::Normal]
+    Normal(&'a std::ffi::OsStr),
+}
+impl PartialOrd for PathComponent<'_> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl Ord for PathComponent<'_> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match (self, other) {
+            (PathComponent::RootDir, PathComponent::RootDir)
+            | (PathComponent::CurDir, PathComponent::CurDir) => Ordering::Equal,
+            (PathComponent::ParentDir(n1), PathComponent::ParentDir(n2)) => n1.cmp(n2).reverse(),
+            (PathComponent::Prefix(c1), PathComponent::Prefix(c2)) => c1.cmp(c2),
+            (
+                PathComponent::Prefix(_),
+                PathComponent::RootDir
+                | PathComponent::ParentDir(_)
+                | PathComponent::CurDir
+                | PathComponent::Normal(_),
+            ) => Ordering::Less,
+            (
+                PathComponent::RootDir
+                | PathComponent::ParentDir(_)
+                | PathComponent::CurDir
+                | PathComponent::Normal(_),
+                PathComponent::Prefix(_),
+            ) => Ordering::Greater,
+            (
+                PathComponent::RootDir,
+                PathComponent::CurDir | PathComponent::ParentDir(_) | PathComponent::Normal(_),
+            ) => Ordering::Less,
+            (PathComponent::ParentDir(_), PathComponent::CurDir | PathComponent::Normal(_)) => {
+                Ordering::Less
+            }
+            (PathComponent::CurDir | PathComponent::Normal(_), PathComponent::ParentDir(_)) => {
+                Ordering::Greater
+            }
+            (
+                PathComponent::CurDir | PathComponent::ParentDir(_) | PathComponent::Normal(_),
+                PathComponent::RootDir,
+            ) => Ordering::Greater,
+            (PathComponent::CurDir, PathComponent::Normal(_)) => Ordering::Less,
+            (PathComponent::Normal(_), PathComponent::CurDir) => Ordering::Greater,
+            (PathComponent::Normal(s1), PathComponent::Normal(s2)) => {
+                ImportSourceAsciiCollator.cmp_osstr(s1, s2)
+            }
+        }
+    }
+}
+
+/// This type is analog to [std::path::Components] with the following changes:
+/// - The iterator yields [PathComponent] instead of [std::path::Component].
+/// - Consecutive parent directories such as `../..` yields a single [PathComponent::ParentDir] with
+///   the count of parent directories.
+/// - `./..` is normalized to `..`
+struct PathComponents<'a> {
+    inner: std::path::Components<'a>,
+}
+impl<'a> From<&'a Path> for PathComponents<'a> {
```

Comment:
Why don't we use `camino` paths in this file?

Reply:
This was written before the integration of camino. Not sure if this brings value here.

Reply:
You’re right, looks like we don’t get the string values out anymore, so it doesn’t really matter :)